src/app/config.rs

```diff
@@ -103,6 +103,12 @@ pub struct Config {
     /// The vector endpoint to send results to
     pub vector_endpoint: String,
 
+    /// The maximum number of retries to attempt when sending results to vector
+    pub vector_max_retries: Option<u32>,
+
+    /// Whether to retry sending results to vector indefinitely
+    pub retry_vector_errors_forever: bool,
```

Comment:
Seems like we should combine this w/ the other option and make it like `-1` or something maybe?

Reply:
i prefer explicit to implicit, and operator error is handled in the case where someone provides an incorrect config